src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -11,16 +11,44 @@ public static partial class Interlocked
     {
         #region CompareExchange
 
+        [Intrinsic]
+        public static byte CompareExchange(ref byte location1, byte value, byte comparand)
+        {
+#if TARGET_AMD64 || TARGET_ARM64
+            return CompareExchange(ref location1, value, comparand);
+#else
+            return RuntimeImports.InterlockedCompareExchange(ref location1, value, comparand);
+#endif
+        }
+
+        [Intrinsic]
+        public static short CompareExchange(ref short location1, short value, short comparand)
+        {
+#if TARGET_AMD64 || TARGET_ARM64
+            return CompareExchange(ref location1, value, comparand);
+#else
+            return RuntimeImports.InterlockedCompareExchange(ref location1, value, comparand);
+#endif
+        }
+
         [Intrinsic]
         public static int CompareExchange(ref int location1, int value, int comparand)
         {
+#if TARGET_AMD64 || TARGET_ARM64
```

Comment:
```suggestion
#if TARGET_X86 || TARGET_AMD64 || TARGET_ARM64 || TARGET_RISCV64
```
Nit: It would be nice to make these ifdefs match the CoreCLR w/ JIT variant, so that there is no confusion about why they are different.

(I know that native AOT is not - yet - supported on RiscV.)